lib/src/widgets/animated_content_widget.dart

```diff
@@ -1,29 +1,95 @@
+import 'package:animated_book_list/src/src.dart';
 import 'package:flutter/material.dart';
 
 // TODO(any): add documentation here
 ///
-class AnimatedContentWidget extends AnimatedWidget {
+class AnimatedContentWidget extends StatelessWidget {
   ///
   const AnimatedContentWidget({
-    required super.listenable,
-    required this.content,
+    required this.bookAnimation,
+    required this.delegate,
     super.key,
   });
 
   ///
-  final Widget content;
+  final Animation<double> bookAnimation;
 
   ///
-  Animation<double> get contentAnimation => listenable as Animation<double>;
+  final AnimatedContentDelegate delegate;
 
   @override
   Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: bookAnimation,
+      builder: (context, child) =>
+          delegate.build(context, bookAnimation, child),
+      child: delegate.contentChild,
+    );
+  }
+}
+
+///
+abstract class AnimatedContentDelegate {
```

Comment:
can you move this class and the implementation classes into a new file? 'animated_content_delegate.dart' file

Reply:
Sure. In which directory should I put it?

Reply:
same level as "animated_content_widget.dart"

Reply:
Done

Reply:
wait, are you using two github accounts? xd

Reply:
haha, Yes I have two accounts. I made a mistake